### ISSUE TITLE
Feature Request & RFC: Ability to set maximum concurrency limit per-pipeline in pipeline configuration.

### ISSUE THREAD

**Issue by THD-Alex-Lowe** [spinnaker/spinnaker#6558]
### Issue Summary

My team is maintaining a Spinnaker instance for use by multiple teams. We’re currently facing an unexpected use case, where specific users need to be able to limit their pipelines so they do not accidentally DDoS downstream services if pipelines are executed in parallel. They are running several hundred pipelines that might end up waiting hours for completion. We can scale very well, so we are being limited by external services.

Ideally, there’s a solution that allows them to restrict or limit pipeline executions on a per-pipeline basis. We could set a global limit on concurrent pipelines, but then they would inadvertently end up blocking all other pipelines while they are executing, which is not ideal.

Another issue previously discussed a similar idea in issue https://github.com/spinnaker/spinnaker/issues/5775, but it was a more general suggestion rather than a specific request, and our situation has moved from “it would be nice” to “essential”.

### Cloud Provider(s)

Any

### Feature Area

Pipeline Configuration

### Description

As a user of Spinnaker Pipelines, I would like to have the ability to limit my specific pipeline from executing more than a certain number of instances in parallel by using a setting similar to the already available `limitConcurrent` setting.

**Comment by guido9j** [comment]
Created draft pull request. We will be discuss this at the Platform SIG this Thursday, November 18.

**Comment by guido9j** [comment]
@link108 : All the 4 PRs my team submitted for this change are ready for review. Please let us know if we need to change anything. Thanks

**Comment by link108** [comment]
@guido9j Thank you for all the PR's 🙏 Everything has been merged, I'm going to close this issue, but please feel to re-open, if needed.
